### DISCUSSION TITLE
Using wavemap without ROS

### DISCUSSION THREAD

**Discussion by Nimaro76** [ethz-asl/wavemap#60] [Q&A] [answered]
Hello, 
do you have a usage example or file that can guide how to use wavemap directly in c++ without using ROS ? 

kind regards,

**Comment by victorreijgwart** [comment]
Hi Nimaro,
How would you like to use it?
The [usage examples](https://ethz-asl.github.io/wavemap/pages/usage_examples.html) in our documentation show how you can load maps from disk and work with them, which doesn't require ROS. We provide example maps that can be downloaded from [here](https://ethz-asl.github.io/wavemap/pages/demos.html). It's also possible to build your own maps without ROS by calling the wavemap library directly, for example like in [this discussion](https://github.com/ethz-asl/wavemap/discussions/53).
In terms of building the code, you could use the [Docker image](https://ethz-asl.github.io/wavemap/pages/installation.html#docker) we provide as an easy way to get started. It still uses catkin (ROS' build system) to build wavemap under the hood, but it's automated and you wouldn't need to install anything on your own machine. Building wavemap yourself without catkin would require writing your own CMake file. If you go that route I can share some pointers, but we don't have concrete usage examples yet.

**Comment by victorreijgwart** [comment, marked as answer]
To anyone reading this, wavemap version 2.0 introduced support for using the framework as a standalone CMake C++ Library. We provide instructions on how to use it in [this tutorial](https://ethz-asl.github.io/wavemap/pages/tutorials/cpp.html). Alternatively, it's now also possible to use wavemap directly [in Python](https://ethz-asl.github.io/wavemap/pages/tutorials/python.html).
